fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/query/WeightEvaluationQuery.java

```diff
@@ -200,7 +205,9 @@ private static DataSource makeInnerQuery(DruidAggregationQuery<?> query, double
                         aggregations,
                         postAggregations,
                         innerQuery.getIntervals(),
-                        ((GroupByQuery) innerQuery).getLimitSpec()
+                        ((GroupByQuery) innerQuery).getLimitSpec() == null ?
+                                null :
+                                ((GroupByQuery) innerQuery).getLimitSpec().withColumns(new LinkedHashSet<>())
```

Comment:
Seeing as how we're doing this twice, and all the casting makes for a fairly complex statement, I would suggest extracting this into a helper method:

```java
private LimitSpec extractLimitSpec(GroupByQuery query) {
   return query.getLimitSpec() == null ? null : query.getLimitSpec().withColumns(new LinkedHashSet());
} 
```
This will simplify the constructor call to:
```java
   ...
   query.getQueryType() == QueryType.GROUP_BY ? extractLimitSpec((GroupByQuery) query) : null
```
and this code to:

```java
...
extractLimitSpec((GroupByQuery) innerQuery)
```

It also means that we can have a positive condition, which is always nice.